src/Power.cpp

```diff
@@ -379,7 +385,15 @@ class AnalogBatteryLevel : public HasBatteryLevel
 #endif
 #ifdef EXT_CHRG_DETECT
         return digitalRead(EXT_CHRG_DETECT) == ext_chrg_detect_value;
+
+#elif HAS_TELEMETRY && !defined(ARCH_PORTDUINO) && !defined(HAS_PMU) && !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR
+        if (hasINA()) {
+            return (getINACurrent() > 0) ? OptTrue : OptFalse;
+        } else {
+            return OptFalse;
+        }
```

Comment:
I think the logic is sound; the larger code block looks like this;

```c
#elif HAS_TELEMETRY && !defined(ARCH_PORTDUINO) && !defined(HAS_PMU) && !MESHTASTIC_EXCLUDE_ENVIRONMENTAL_SENSOR
        if (hasINA()) {
            return (getINACurrent() > 0) ? OptTrue : OptFalse;
        } else {
            return OptFalse;
        }
#else

        return isBatteryConnect() && isVbusIn();
#endif
```

What I can do instead is remove both the `else` and the `#else` so that even if we SHOULD have an INA and don't, it still falls through and returns `isBatteryConnect() && isVbusIn()` -- I think I like that logic better, thanks for pointing that out!

Reply:
In the original version, I think most devices which do not have an INA sensor would have always returned OptFalse there, but looks like your changes have got it all sorted out now!